### PR TITLE
ACT: Try to group auto-inserted use item with existing ones by last component

### PR DESCRIPTION
**Pull request by mchernyavsky**
Closes https://github.com/intellij-rust/intellij-rust/issues/3403.

### CODE REVIEW

**Comment by vlad20012** [comment]
_Really_ is should be done via "Optimize imports" action.

Also, should it be optional?

**Comment by Undin** [comment]
> _Really_ is should be done via "Optimize imports" action.

Looks like no. "Optimize import" action usually reorders imports, removes unused, etc. I think it's unexpected during completion. For example, kotlin implementation of completion from index/auto import just insert new `import` item without additional actions

File: src/test/kotlin/org/rust/ide/inspections/import/AutoImportFixTest.kt [hunk 1]
```diff
@@ -1598,4 +1604,160 @@ class AutoImportFixTest : AutoImportFixTestBase() {
...
+            pub struct Foo;
+            pub struct Bar;
+        }
+
+        fn main() {
+            let f = <error descr="Unresolved reference: `Bar`">Bar/*caret*/</error>;
+        }
+    """, """
+        use foo::Foo as F;
+        use foo::Bar;
```
Comment (Undin) [inline src/test/kotlin/org/rust/ide/inspections/import/AutoImportFixTest.kt]: Are you sure that we don't want to merge use items in this case?
`use foo::{Foo as F, Bar}` is valid and semantically the same

Comment (mchernyavsky) [reply src/test/kotlin/org/rust/ide/inspections/import/AutoImportFixTest.kt]: Fixed.

**Comment by vlad20012** [comment]
Comments above use items should be preserved:
peek 2019-02-23 22-56

File: src/main/kotlin/org/rust/ide/inspections/import/AutoImportFix.kt [hunk 2]
```diff
@@ -505,6 +506,49 @@ private fun RsMod.insertUseItem(psiFactory: RsPsiFactory, usePath: String) {
...
 }
 
+private fun RsMod.tryMergeWithOtherUseItems(psiFactory: RsPsiFactory, newUseItem: RsUseItem): Boolean =
+    childrenOfType<RsUseItem>().any { newUseItem.tryMergeWith(psiFactory, it) }
+
+private fun RsUseItem.tryMergeWith(psiFactory: RsPsiFactory, other: RsUseItem): Boolean {
+    if (other.vis != null || other.outerAttrList.isNotEmpty() || other.useSpeck?.isStarImport == true) return false
+
+    val parentPath = parentPath ?: return false
+    val importingName = importingNames?.singleOrNull() ?: return false
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/inspections/import/AutoImportFix.kt]: parentPath and importingName is computed multiple times for "this" when could be computed once

**Comment by mchernyavsky** [comment]
Fixed.

**Pull request merged**